src/main/java/com/whichclasses/frontend/SearchApiServlet.java

```diff
@@ -29,8 +35,31 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response)
       throws ServletException, IOException {
     response.setStatus(HttpServletResponse.SC_OK);
     response.setContentType("application/json");
-    // TODO: actually implement this
-    response.getWriter().write("{\"results\":[{\"identifier\":{\"department\":\"ECE\",\"course_number\":\"351A\"},\"title\":\"All The Circuits\"}]}");
+    String quality = request.getParameter("quality");
+    String entityType = request.getParameter("entityType");
+    String department = request.getParameter("department");
+    
+    Question question = Question.OVERALL_COURSE_RATING;
+    switch(quality) {
+    case "best":
+      question = Question.OVERALL_COURSE_RATING;
+      break;
+    case "hardest":
+      question = Question.COURSE_DIFFICULTY;
+      break;
+    }
+    
+    String json = "{\"results\":[";
+    List<Course> courses = mDataAggregator.getRankedCourses(department, question);
+    for(Course c : courses) {
+    	String[] split = c.toString().split(" - ", 2);
+    	String number = split[0];
+    	String title = split[1];
+    	// It works.
```

Comment:
Look into GSON. Used a few other places in the code already. Don't write the JSON yourself.